Documentation/disaster-recovery.md

```diff
@@ -106,3 +106,102 @@ kubectl create -f operator.yaml
+
+In author's situation,  `etcd` data of running Kubernetes cluster is nuked and have no backup, all OSDs are using bluestore backend.
+
+### Prerequisites
+
+1. A working Kubernetes cluster without rook
+2. Previous rook/ceph cluster is intact, which means you have at least one ceph mon data intact, and all your OSD data is intact.
+
+Tested in `rook/ceph:v1.0.3` and `ceph/ceph:v14.2.1-20190430`.
+
+### TL;DR for Steps below
+1. Start a new and clean rook cluster, with old `CephCluster` & friends. Shut it down when it seems working(as a brand new cluster).
+3. Replace ceph-mon data with old one. Fix `fsid` in rook. Fix monmap. Fix mon auth. Disable auth.
+4. Spin it up, watch it resurrect. Fix admin auth key.
+5. Shut it down again. Enable auth. Spin it up.
+
+HOORAY!
+
+### Steps
+
+0. Assuming your old Kubernetes cluster is completely torned down, and your new Kubernetes cluster is up and running, without rook.
+1. Backup `/var/lib/rook` in all your rook nodes. Backups will be used later.
+2. Pick a `/var/lib/rook/rook-ceph/rook-ceph.config`  from any node and get your old cluster `fsid` from its content.
+3. Remove `/var/lib/rook` in all your rook nodes.
+4. Install rook in your new Kubernetes cluster.
+5. Prepare identical `CephCluster`  descriptors, especially identical `spec.storage.config` and `spec.storage.nodes`,  except `mon.count`, which sets to `1`. Post them to your new Kubernetes cluster.
+6. Prepare identical `CephFilesystem` & ... etc descriptors (if any). Post them to your new Kubernetes cluster too.
+7. Run `kubectl logs -f rook-ceph-operator-xxxxxxxxxx` and wait till all the things are settled.
+8. Run `kubectl get cm/rook-crush-config -o yaml`, ensure `initialCrushMapCreated` is set to `1`. If not, goto 7, manually set it or stop here for further help.
+9. **STATE:** Now you will have `rook-ceph-mon-a`, `rook-ceph-mgr-a`, and all the auxiliary pods up and running, and zero(hopefully) `rook-ceph-osd-X` running. Rook should not start any OSD daemon since all devices belongs to your old cluster(have a different `fsid`).
+10. Run `kubectl exec -it rook-ceph-mon-a-XXXXXX bash` to enter your `ceph-mon` pod,
+
+```shell
+mon-a# cat /etc/ceph/keyring-store/keyring  # save this keyring content, for later use
+mon-a# exit
+```
+
+11. Run `kubectl edit deploy/rook-ceph-operator` and set `replicas` to 0.
+12. Run `kubectl delete deploy/X` where X is every deployment in namespace `rook-ceph`, except `rook-ceph-operator` and `rook-ceph-tools`.
+
+SSH to the host where `rook-ceph-mon-a` in your new Kubernetes cluster resides.
+
+13. Pick the latest `ceph-mon` directory (`/var/lib/rook/mon-?`) in your previous backup, replace `/var/lib/rook/mon-a` with it.
+14. Replace `/var/lib/rook/mon-a/keyring` with the saved keyring, preserving only the `[mon.]` section, remove `[client.admin]` section.
+15. Get your `rook-ceph-mon-a` address by `kubectl get cm/rook-ceph-mon-endpoints -o yaml` in your new Kubernetes cluster.
```

Comment:
This is not a step which needs to be done from the mon node. Move it before "SSH to the host where `rook-ceph-mon-a` in your new Kubernetes cluster resides.", and replace "Get" again with "Note."

Reply:
fixed